pkg/network/discovery.go

```diff
@@ -34,6 +35,7 @@ type DefaultDiscovery struct {
 	connectedAddrs   map[string]bool
 	goodAddrs        map[string]bool
 	unconnectedAddrs map[string]int
+	isDead           bool
```

Comment:
Not the most elegant solution, but with the current design I'm not sure anything better is possible (absent this kludge we'll panic on `RequestRemote` after `Close`).